docs/DafnyRef/Statements.md

```diff
@@ -1739,8 +1746,92 @@ RevealStmt =
     ";"
 ````
 
+The `reveal` stastement makes available to the solver information that is otherwise opaque (that is, not visible).
+
+### 20.20.1. Revealing assertions
+
+If an assert statement has an expression label, then a proof of that assertion is attempted, but the assertion itself
+is not used subsequently.  For example, consider
+```dafny
+method m(i: int) {
+  assert x: i == 0; // Fails
+  assert i == 0; // Fails also because the x: makes the first assertion opaque
+}
+```
+The first assertion fails. Without the label `x:`, the second would succeed because after a failing assertion, the 
+assertion is assumed in the context of the rest of the program.  But with the label, the first assertion is hidden from
+the rest of the program. That assertion can be _revealed_ by adding a `reveal` statement:
+
+```dafny
+method m(i: int) {
+  assert x: i == 0; // Fails
+  reveal x;
+  assert i == 0; // Now succeeds
```

Comment:
Likewise to your example of hiding preconditions, can you use `assert i == 0 by { reveal x; }` or describe that it is possible to scope the reveal using this method?